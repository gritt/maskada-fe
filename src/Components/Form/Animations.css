.animation-place {
    display: flex;
}

.success {
    box-shadow: 0 0 100px var(--color-success);
}

.success div {
    box-shadow: 0 0 130px var(--color-success);
}

.loading {
    box-shadow: 0 0 100px var(--color-info);
}

.loading div {
    box-shadow: 0 0 130px var(--color-info);
}

.animation {
    z-index: -1;

    width: 8rem;
    height: 8rem;
    margin: 0 auto;
    position: relative;

    border-radius: 100%;
    overflow: hidden;
}

.animation div {
    position: absolute;
    width: 4rem;
    height: 4rem;
    border-radius: 100%;

    animation: processing 0.5s linear infinite;
}

.animation div:nth-child(1) {
    top: 8px;
    left: 8px;
    animation-delay: 0s;
}

.animation div:nth-child(2) {
    top: 8px;
    left: 32px;
    animation-delay: -0.4s;
}

.animation div:nth-child(3) {
    top: 8px;
    left: 56px;
    animation-delay: -0.8s;
}

.animation div:nth-child(4) {
    top: 32px;
    left: 8px;
    animation-delay: -0.4s;
}

.animation div:nth-child(5) {
    top: 32px;
    left: 32px;
    animation-delay: -0.8s;
}

.animation div:nth-child(6) {
    top: 32px;
    left: 56px;
    animation-delay: -1.2s;
}

.animation div:nth-child(7) {
    top: 56px;
    left: 8px;
    animation-delay: -0.8s;
}

.animation div:nth-child(8) {
    top: 56px;
    left: 32px;
    animation-delay: -1.2s;
}

.animation div:nth-child(9) {
    top: 56px;
    left: 56px;
    animation-delay: -1.6s;
}

@keyframes processing {
    0%, 100% {
        opacity: 1;
    }
    50% {
        opacity: 0.5;
    }
}
