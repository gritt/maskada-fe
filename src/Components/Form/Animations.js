import React from "react";
import "./Animations.css"

const Animation = (className) => {
    let style = `animation ${className}`

    return (
        <div className={style}>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
        </div>
    )

}

const Loading = () => {
    return (
        <div className={"animation-place"}>
            {Animation('loading')}
        </div>
    )
}

const Success = () => {
    return (
        <div className={"animation-place"}>
            {Animation('success')}
        </div>
    )
}

export {Loading, Success}
