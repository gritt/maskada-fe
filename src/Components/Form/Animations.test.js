import React from "react";
import {render,} from '@testing-library/react'
import {Loading, Success} from "./Animations";

describe('Animations', () => {
    it('renders loading', () => {
        // when
        let {container} = render(<Loading/>)

        // then
        expect(container).toMatchSnapshot()
    });
    it('renders success', () => {
        // when
        let {container} = render(<Success/>)

        // then
        expect(container).toMatchSnapshot()
    });
})
