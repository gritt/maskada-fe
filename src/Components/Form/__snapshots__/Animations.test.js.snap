// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animations renders loading 1`] = `
<div>
  <div
    class="animation-place"
  >
    <div
      class="loading"
    />
  </div>
</div>
`;

exports[`Animations renders success 1`] = `
<div>
  <div
    class="animation-place"
  >
    <div
      class="success"
    >
      ✓︎
    </div>
  </div>
</div>
`;
